iceoryx_hoofs/source/posix_wrapper/file_lock.cpp

```diff
@@ -90,13 +90,13 @@ FileLock& FileLock::operator=(FileLock&& rhs) noexcept
                       << "\" in the move constructor/move assingment operator" << std::endl;
         }
 
-        CreationPattern_t::operator=(std::move(rhs));
-
         m_name = std::move(rhs.m_name);
-        m_fd = std::move(rhs.m_fd);
+        m_fd = rhs.m_fd;
```

Comment:
please revert and check that all the code changes doing similar stuff is also reverted